examples/integration/packages/my-integration/src/integration.ts

```diff
@@ -0,0 +1,37 @@
+import type { AstroIntegration } from 'astro';
+import pkg from '../package.json';
+
+export default function createIntegration(): AstroIntegration {
+	return {
+		name: pkg.name,
+		hooks: {
+			'astro:config:setup': ({
```

Comment:
Thought it would be nice to stub in details for the most commonly used hooks

Reply:
This fits in great as an empty starter, but I wonder if a minimal working example/demo might be especially helpful here, since it's hard to understand how to practically use each hook. We could fill in some or all of the hooks show you can use the arguments passed to each hook.

Reply:
Yeah I was a little torn on this one as well.  I originally had every hook stubbed in here as a comment with a note linking to the docs for more details

I couldn't think of a good example for the hooks that didn't become more opinionated but I'll take another stab at it

Reply:
I think it's fine to just add links to examples of these hooks being used by other integrations

Reply:
Yep, that's a great idea!  I updated the integration skeleton to links for the API docs and a few example hooks in our @astrojs integrations